tests/AuthenticationProviderTests.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+#include "AuthenticationProvider.hpp"
+#include "User.hpp"
+#include <string>
+
+
+TEST(AuthenticationProviderTests, canAuthenticate)
+{
+    User user("Alicja", "AJU", "wieczorowa", "alicjaliQui", "alicjaliQui", "alicjaliqui@gmail.com", "lubieplacki");
+    AuthenticationProvider authenticator;
+
+    std::string email = "alicjaliqui@gmail.com";
+    std::string password = "lubieplacki";
+
+    EXPECT_EQ(authenticator(user, email, password), true);
+}
+
+TEST(AuthenticationProviderTests, canAuthenticateEmailCaseInsensitivity)
+{
+    User user("Alicja", "AJU", "wieczorowa", "alicjaliQui", "alicjaliQui", "alicjaliqui@gmail.com", "lubieplacki");
+    AuthenticationProvider authenticator;
+
+    std::string email = "alicjaliqui@gmail.com";
+    std::string password = "lubieplacki";
+
+    EXPECT_EQ(authenticator(user, email, password), true);
+}
+
+TEST(AuthenticationProviderTests, canNotAuthenticateWrongPassword)
+{
+    User user("Alicja", "AJU", "wieczorowa", "alicjaliQui", "alicjaliQui", "alicjaliqui@gmail.com", "lubieplacki");
+    AuthenticationProvider authenticator;
+
+    std::string email = "alicjaliqui@gmail.com";
+    std::string password = "lubiePLACKI";
+
+    EXPECT_EQ(authenticator(user, email, password), false);
+}
+
+TEST(AuthenticationProviderTests, canNotAuthenticateEmptyPassword)
```

Comment:
the test name says canNotAuthenticate, but it is authenticated with empty password.